### DISCUSSION TITLE
How to check if cancel button is pressed in dialog

### DISCUSSION THREAD

**Discussion by hepabolu** [ttscoff/bunch-docs#278] [Q&A] [answered]
I want Bunch to test if the cancel button is pressed. If so, exit the Bunch. However, I cannot get it to work. This is the script:

```
project=?{
    Project A => P1
    Project B => P2
    Project C => P2
} "Which project"

(log project is set to ${project} )

# if a project is selected
if ${project} not equals ''
    (notify project ${project} is selected)
else
	(notify dialog cancelled)
	# $ exit 0
end
```

Whatever I select, the else part not reached. I have tried various if conditions, but I cannot find the proper way to distinguish between a selected project and a cancel.

How should the condition be?

**Comment by hepabolu** [comment, marked as answer]
Looks like the if statement should be:

```
# if no project is selected
if !project
	(notify dialog cancelled)
	# $ exit 0
else
    (notify project ${project} is selected)
end
```

**Reply by ttscoff** [reply]
Thanks, will add this to the docs if I can find the right place for it :)

**Reply by ttscoff** [reply]
If I'm reading my own code correctly, cancelling a dialog does actually fail to set the variable, so comparison to an empty string would still return false. Negating the variable is the correct method, (could also be written as `if not project`).
